engine/src/main/java/org/pentaho/di/trans/steps/propertyoutput/PropertyOutput.java

```diff
@@ -0,0 +1,286 @@
+          openNewFile();
+        }
+      }
+
+      if ( !data.KeySet.contains( propkey ) ) {
+        if ( log.isDetailed() ) {
+          logDetailed( BaseMessages.getString( PKG, "PropertyOutput.Log.Key", propkey ) );
+          logDetailed( BaseMessages.getString( PKG, "PropertyOutput.Log.Value", propvalue ) );
+        }
+        // Update property
+        data.pro.setProperty( propkey, propvalue );
+        putRow( data.outputRowMeta, r ); // in case we want it to go further...
+        incrementLinesOutput();
+
+        if ( checkFeedback( getLinesRead() ) ) {
+          if ( log.isBasic() ) {
+            logBasic( "linenr " + getLinesRead() );
+          }
+        }
+        data.KeySet.add( propkey );
+      }
+    } catch ( KettleStepException e ) {
+      boolean sendToErrorRow = false;
+      String errorMessage = null;
+      if ( getStepMeta().isDoingErrorHandling() ) {
+        sendToErrorRow = true;
+        errorMessage = e.toString();
+      } else {
+        logError( BaseMessages.getString( PKG, "PropertyOutputMeta.Log.ErrorInStep" ) + e.getMessage() );
+        setErrors( 1 );
+        stopAll();
+        setOutputDone(); // signal end to receiver(s)
+        return false;
+      }
+      if ( sendToErrorRow ) {
+        putError( data.outputRowMeta, r, 1L, errorMessage, null, "PROPSOUTPUTO001" );
+      }
+    }
+    return true;
+  }
+
+  public boolean checkSameFile() throws KettleException {
+    return data.previousFileName.equals( data.filename );
+  }
+
+  private void openNewFile() throws KettleException {
+    try ( FileObject newFile = KettleVFS.getFileObject( data.filename, getTransMeta() ) ) {
+      data.pro = new Properties();
+      data.KeySet.clear();
+
+      data.file = newFile;
+      if ( meta.isAppend() && data.file.exists() ) {
+        data.pro.load( KettleVFS.getInputStream( data.file ) );
+      }
+      // Create parent folder if needed...
+      createParentFolder();
+      //save processing file
+      data.previousFileName = data.filename;
+    } catch ( Exception e ) {
+      throw new KettleException( "Error opening file [" + data.filename + "]!", e );
+    }
+  }
+
+  private void createParentFolder() throws KettleException {
+    if ( meta.isCreateParentFolder() ) {
+      FileObject parentfolder = null;
+      try {
+        // Do we need to create parent folder ?
+
+        // Check for parent folder
+        // Get parent folder
+        parentfolder = data.file.getParent();
+        if ( !parentfolder.exists() ) {
+          if ( log.isDetailed() ) {
+            logDetailed( BaseMessages.getString( PKG, "PropertyOutput.Log.ParentFolderExists", parentfolder.getName().toString() ) );
+          }
+          parentfolder.createFolder();
+          if ( log.isDetailed() ) {
+            logDetailed( BaseMessages.getString( PKG, "PropertyOutput.Log.CanNotCreateParentFolder", parentfolder.getName().toString() ) );
+          }
+        }
+      } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 221.